src/components/MediatedRequestsActivities/MediatedRequestsActivities.js

```diff
@@ -34,15 +41,159 @@ import {
   MEDIATED_REQUESTS_RECORD_TRANSLATIONS,
   ICONS,
+  SEARCH_FIELDS,
+  FILTER_CONFIG,
   getMediatedRequestsActivitiesUrl,
 } from '../../constants';
-import { getTotalCount } from '../../utils';
+import {
+  getTotalCount,
+  getDeliveryAddressForCsvRecords,
+  getFullNameForCsvRecords,
+} from '../../utils';
+
+const REPORT_HEADERS = [
+  'requestType',
+  'status',
+  'requestDate',
+  'item.barcode',
+  'instance.title',
+  'instance.contributorNames',
+  'item.location.libraryName',
+  'item.callNumberComponents.prefix',
+  'item.callNumberComponents.callNumber',
+  'item.callNumberComponents.suffix',
+  'item.volume',
+  'item.enumeration',
+  'item.chronology',
+  'item.copyNumber',
+  'requester.name',
+  'requester.barcode',
+  'requester.patronGroup.group',
+  'fulfillmentPreference',
+  'pickupServicePoint.name',
+  'deliveryAddress',
+  'proxy.name',
+  'proxy.barcode',
+  'patronComments',
+];
 
-export const getActionMenu = (renderColumnsMenu, history) => () => {
+export const getActionMenu = ({
+  renderColumnsMenu,
+  mediatedRequests,
+  searchValue,
+  activeFilters,
+  reportRecords,
+  callout,
+  history,
+  isLoadingReport,
+  setIsLoadingReport,
+  formatMessage,
+  // eslint-disable-next-line react/prop-types
+}) => ({ onToggle }) => {
   const goToNewMediatedRequest = () => {
     history.push(`${getMediatedRequestsActivitiesUrl()}/create`);
   };
 
+  const getRecords = async (query) => {
+    const {
+      GET,
+      reset,
+    } = reportRecords;
+
+    const limit = 1000;
+    const data = [];
+    let offset = 0;
+    let hasData = true;
+
+    while (hasData) {
+      try {
+        reset();
+        // eslint-disable-next-line no-await-in-loop
+        const result = await GET({
+          params: {
+            query,
+            limit,
+            offset,
+          },
+        });
+        hasData = result.length;
+        offset += limit;
+
+        if (hasData) {
+          data.push(...result);
+        }
+
+        if (result.length < limit) {
+          hasData = false;
+        }
+      } catch (err) {
+        hasData = false;
+      }
+    }
+
+    return data;
+  };
+
+  const getColumnHeaders = (headers) => {
+    return headers.map(header => ({
+      label: formatMessage({ id: `ui-requests-mediated.report.${header}` }),
+      value: header,
+    }));
+  };
+
+  const buildRecords = (records) => {
+    const clonedRecords = cloneDeep(records);
+
+    clonedRecords.forEach(record => {
+      if (record.instance.contributorNames?.length > 0) {
+        record.instance.contributorNames = record.instance.contributorNames.map(({ name }) => name).join('; ');
+      }
+
+      if (record.proxy) {
+        record.proxy.name = getFullNameForCsvRecords(record.proxy);
+      }
+
+      if (record.deliveryAddress) {
+        record.deliveryAddress = getDeliveryAddressForCsvRecords(record.deliveryAddress);
+      }
+
+      record.requester.name = getFullNameForCsvRecords(record.requester);
```

Comment:
Could you please destructure record for instance, proxy, deliveryAddress and others?

Reply:
Fixed